aptos-node/src/lib.rs

```diff
@@ -301,14 +328,74 @@ where
     start(config, Some(log_file), false)
 }
 
+/// Merge two YAML values, useful for merging NodeConfigs
+fn merge_yaml(a: &mut serde_yaml::Value, b: serde_yaml::Value) {
+    match (a, b) {
+        (a @ &mut serde_yaml::Value::Mapping(_), serde_yaml::Value::Mapping(b)) => {
+            let a = a.as_mapping_mut().unwrap();
+            for (k, v) in b {
+                if v.is_sequence() && a.contains_key(&k) && a[&k].is_sequence() {
+                    let mut _b = a.get(&k).unwrap().as_sequence().unwrap().to_owned();
+                    _b.append(&mut v.as_sequence().unwrap().to_owned());
+                    a[&k] = serde_yaml::Value::from(_b);
+                    continue;
+                }
+                if !a.contains_key(&k) {
+                    a.insert(k.to_owned(), v.to_owned());
+                } else {
+                    merge_yaml(&mut a[&k], v);
+                }
+            }
+        },
+        (a, b) => *a = b,
```

Comment:
does this behavior make sense?

Or should we error on merge of things which are not mapping?

Also you can merge sequence, although not sure if you want to do that at a high level